lib/async.js

```diff
@@ -0,0 +1,126 @@
+'use strict';
+
+const LRU = require('lru-cache');
+const hash = require('object-hash');
+
+class LockingAsync {
+  /*
+  temporary locker for holding resolve/rejects of
+  concurrent cache misses for the same key
+
+  The locker key includes an array of promise resolve/reject
+  functions to be invoked once the original caller
+  fetches the result of the operation. Example:
+
+  {
+    "1119717257625be99": [[resolve, reject], [resolve, reject], [resolve, reject]]
+  }
+  */
+  #locks;
+  #stale;
+  #stats;
+
+  constructor(func, params) {
+    if (!func) throw new Error('Locking: locking class requires two arguments: function, params');
+    this.func = func;
+    this.cache = new LRU({ ...{ max: 1000 }, ...params });
```

Comment:
Is destructuring `...{ max: 100 }` for some specific reason, wouldn't `new LRU({ max: 1000, ...params });` result with the same options object for LRU ?

Reply:
Yep, looks like I accidentally doubled up the objects here. I'll simplify, thanks! Resolved in 2da4640